addons/hr_expense/models/hr_expense.py

```diff
@@ -1294,115 +1414,42 @@ def _do_create_moves(self):
         own_account_sheets = self.filtered(lambda sheet: sheet.payment_mode == 'own_account')
         company_account_sheets = self - own_account_sheets
 
-        moves = self.env['account.move'].create([sheet._prepare_bill_vals() for sheet in own_account_sheets])
-        payments = self.env['account.payment'].with_context(**skip_context).create([sheet._prepare_payment_vals() for sheet in company_account_sheets])
-        moves |= payments.move_id
+        moves = self.env['account.move'].create(own_account_sheets._prepare_bills_vals())
+        payments = self.env['account.payment'].with_context(**skip_context).create(company_account_sheets.mapped('expense_line_ids')._prepare_payments_vals())
+        payment_moves = payments.move_id
+        payment_moves_baselines = payment_moves.line_ids.filtered(lambda line: line.display_type == 'product')
+        for line in payment_moves_baselines:
+            # We need to compute the price_total and price_subtotal of the baselines properly, else the unreconciliation fails
+            line.with_context(**skip_context).write({'price_subtotal': line.amount_currency, 'price_total': line.move_id.amount_total})
```

Comment:
You need another way to do that. I don't get the problem when unreconciling. To be discussed